google/cloud/spanner/internal/time_format_test.cc

```diff
@@ -27,13 +27,86 @@ constexpr auto kTimeFormat = "%Y-%m-%dT%H:%M:%S";
 
 TEST(TimeFormat, Format) {
   std::tm tm;
+
   tm.tm_year = 2019 - 1900;
   tm.tm_mon = 6 - 1;
   tm.tm_mday = 21;
   tm.tm_hour = 16;
   tm.tm_min = 52;
   tm.tm_sec = 22;
   EXPECT_EQ("2019-06-21T16:52:22", FormatTime(kTimeFormat, tm));
+
+#if !defined(_WIN32)
+  tm.tm_year = 10000 - 1900;
+  tm.tm_mon = 6 - 1;
+  tm.tm_mday = 21;
+  tm.tm_hour = 16;
+  tm.tm_min = 52;
+  tm.tm_sec = 22;
+  EXPECT_EQ("10000-06-21T16:52:22", FormatTime(kTimeFormat, tm));
+#endif
+
+  tm.tm_year = 123 - 1900;
+  tm.tm_mon = 6 - 1;
+  tm.tm_mday = 21;
+  tm.tm_hour = 16;
+  tm.tm_min = 52;
+  tm.tm_sec = 22;
+#if defined(__APPLE__) || defined(_WIN32)
+  EXPECT_EQ("0123-06-21T16:52:22", FormatTime(kTimeFormat, tm));
```

Comment:
> Do we know if Cloud Spanner accepts both formats?

I'll say it doesn't matter because, in the case of Spanner, we don't use the general form of `FormatTime()` (i.e., the one with a `fmt` string), so we're are not subject to the `"%Y"` vagaries.